pkg/psmdb/backup/backup.go

```diff
@@ -60,13 +74,29 @@ func HasActiveJobs(cl client.Client, cluster, namespace string, current Job) (bo
 		if r.Name == current.Name && current.Type == TypeRestore {
 			continue
 		}
-		if r.Spec.ClusterName == cluster &&
+		if r.Spec.ClusterName == cluster.Name &&
 			r.Status.State != api.RestoreStateReady &&
 			r.Status.State != api.RestoreStateError &&
 			r.Status.State != api.RestoreStateWaiting {
 			return true, nil
 		}
 	}
 
+	pbm, err := NewPBM(cl, cluster)
+	if err != nil {
+		return false, errors.Wrap(err, "getting pbm object")
+	}
+
+	allowLock = append([]LockHeaderPredicate{NotJobLock(current)}, allowLock...)
```

Comment:
could be simplified as
`allowLock = append(allowLock,NotJobLock(current))`

Reply:
updated in 2ab5eebbccd167792aea4b69b5fae6680f19e217